### PR TITLE
update (py)cryptominisat to version 5.11.21, remove no longer needed cryptominisat spkg

### LINKED ISSUE #34687: Upgrade: cryptominisat 5.11
<div id="comment:0"></div>

CryptoMiniSat 5.11 is out.

Our last upgrade was

- #25374: Upgrade cryptominisat to 5.8.0, fix build on Cygwin

CC:  @kiwifb @slel

Component: **packages: optional**

Keywords: **upgrade, cryptominisat**

_Issue created by migration from https://trac.sagemath.org/ticket/34687_

### PR DESCRIPTION
**Pull request by dimpase**
Fixes #34687
As pycryptosat is now a PyPI package, convert it to a pip package, and remove cryptominisat
### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by dimpase** [comment]
much more economic update - just convert pycryptosat into pip package, and drop cryptominisat as not needed as a standalone

**Comment by mkoeppe** [comment]
You converted it to a non-package. 
Check that you committed all files.

**Comment by dimpase** [comment]
> You converted it to a non-package. Check that you committed all files.

This removes cryptominisat, and only leaves pycryptominisat. We don't need cryptominisat by itself.

**Comment by dimpase** [comment]
> You converted it to a non-package. Check that you committed all files.

PS. As the reviewer is blocking me on GitHub, I don't get any corresponding notifications. 
That's why I sat on this for so long.
If the reviewer wants to proceed in a timely manner, he better appoints a proxy to communicate here.
@saraedum

**Comment by dimpase** [comment]
"failing" incremental CI are actually all good, only after the tests are run and all pass, some rather opaque Docker error pops up
```
[...]
--------------------
ERROR: failed to solve: process "/bin/sh -c make SAGE_SPKG=\"sage-spkg -y -o\" ${USE_MAKEFLAGS} ${TARGETS}" did not complete successfully: exit code: 2
DOCKER_BUILDKIT=1, so we cannot commit and tag the failed image
docker-debian-bullseye-standard-incremental: exit 1 (6748.90 seconds) /home/runner/work/sage/sage> bash -c 'for docker_target in with-targets; do BUILD_IMAGE_STEM=sage-$(echo docker-debian-bullseye-standard-incremental | sed "s/docker-//;s/-incremental//"); BUILD_IMAGE=$DOCKER_PUSH_REPOSITORY$BUILD_IMAGE_STEM-$docker_target; BUILD_TAG=96a1e14a13; TAG_ARGS=$(for tag in $BUILD_TAG 37669-merge; do echo --tag $BUILD_IMAGE:$tag; done); DOCKER_BUILDKIT=1; docker build . -f /home/runner/work/sage/sage/.tox/docker-debian-bullseye-standard-incremental/Dockerfile --target $docker_target $TAG_ARGS --build-arg EXTRA_CONFIGURE_ARGS="--enable-experimental-packages --enable-download-from-upstream-url   --with-system-python3=yes  --enable-fat-binary" --build-arg BASE_IMAGE=ghcr.io/sagemath/sage/sage-$(echo docker-debian-bullseye-standard-incremental | sed -E "s/(docker-|-incremental|-sitepackages)//g")-with-targets:dev --build-arg BOOTSTRAP="./bootstrap" --build-arg TARGETS_PRE="$(if test -n "$TARGETS_PRE"; then echo $TARGETS_PRE; else echo reconfigure build doc-html ptest; fi)" --build-arg TARGETS="reconfigure build doc-html ptest" --build-arg TARGETS_OPTIONAL="build/make/Makefile" --build-arg NUMPROC=9 --build-arg USE_MAKEFLAGS="-k V=0 SAGE_NUM_THREADS=5"; status=$?; unset CONTAINER; if [ $status != 0 ]; then if [ $DOCKER_BUILDKIT = 0 ]; then BUILD_TAG="$BUILD_TAG-failed"; CONTAINER=$(docker ps -l -q); docker commit $CONTAINER $BUILD_IMAGE:$BUILD_TAG; else unset BUILD_TAG; echo "DOCKER_BUILDKIT=1, so we cannot commit and tag the failed image"; fi; fi; if [ -n "$BUILD_TAG" ]; then echo "Copying logs from the container to /home/runner/work/sage/sage/.tox/docker-debian-bullseye-standard-incremental/sage/"; rm -f /home/runner/work/sage/sage/.tox/docker-debian-bullseye-standard-incremental/sage/STATUS; docker run $BUILD_IMAGE:$BUILD_TAG bash -c " tar -c --ignore-failed-read -f - /sage/STATUS /sage/logs /sage/{prefix,venv}/var/lib/sage/installed ~/.sage/timings2.json /sage/pkgs/*/.tox/*/.sage/timings2.json /sage/pkgs/*/.tox/*/logs 2> /dev/null" | (cd /home/runner/work/sage/sage/.tox/docker-debian-bullseye-standard-incremental && tar xf -); if [ -f /home/runner/work/sage/sage/.tox/docker-debian-bullseye-standard-incremental/sage/STATUS ]; then status=$(cat /home/runner/work/sage/sage/.tox/docker-debian-bullseye-standard-incremental/sage/STATUS); fi; fi; unset PUSH_TAGS; if [ -n "$BUILD_TAG" ]; then if [ $status != 0 ]; then BUILD_TAG="${BUILD_TAG%-failed}-failed"; PUSH_TAGS=$BUILD_IMAGE:$BUILD_TAG; else PUSH_TAGS=$(echo $BUILD_IMAGE:$BUILD_TAG; for tag in 37669-merge; do echo "$BUILD_IMAGE:$tag"; done); fi; echo $BUILD_IMAGE:$BUILD_TAG >> /home/runner/work/sage/sage/.tox/docker-debian-bullseye-standard-incremental/Dockertags; fi; if [ x"ghcr.io/sagemath/sage/" != x -a x"$PUSH_TAGS" != x ]; then echo Pushing $PUSH_TAGS; for tag in $PUSH_TAGS; do if docker push $tag; then echo $tag >> /home/runner/work/sage/sage/.tox/docker-debian-bullseye-standard-incremental/Dockertags.pushed; else echo "(ignoring errors)"; fi; done; fi; if [ $status != 0 ]; then exit $status; fi; done' pid=4212
  docker-debian-bullseye-standard-incremental: FAIL code 1 (6750.75=setup[0.87]+cmd[0.98,0.01,6748.90] seconds)
  evaluation failed :( (6750.87 seconds)
Error: Process completed with exit code 1.
```

**Comment by orlitzky** [comment]
> "failing" incremental CI are actually all good, only after the tests are run and all pass, some rather opaque Docker error pops up

this is https://github.com/sagemath/sage/issues/37786

**Comment by mkoeppe** [comment]
> You converted it to a non-package.
> Check that you committed all files.

I'll explain it again: In this PR, the package `pycryptosat`, is a non-package because it has neither a `checksums.ini` (which would make it a normal or wheel package) nor a `requirements.txt` (which would make it a pip package) nor an `spkg-install` (which would make it a script package).

Actually try `make pycryptosat` with your branch. It gives you:
```
[pycryptosat-none] Error: pycryptosat is a dummy package and 
[pycryptosat-none] cannot be installed using the Sage distribution.
```

**Comment by dimpase** [comment]
perhaps I forgot to check in requirements.txt

I'll fix it tomorrow

**Comment by mkoeppe** [comment]
Thanks. Now pycryptosat has the needed `requirements.txt`, but it still has a `version_requirements.txt` (né `install-requires.txt`) file, which should be removed.

**Comment by dimpase** [comment]
that was a fallout of mass file renaming, with  unneeded stuff sneaking in with a merge.

**Comment by mkoeppe** [comment]
> that was a fallout of mass file renaming

No it wasn't.

File: src/sage/sat/solvers/satsolver.pyx [hunk 1]
```diff
@@ -323,7 +323,7 @@ def SAT(solver=None, *args, **kwds):
 
     - ``solver`` (string) -- select a solver. Admissible values are:
 
-        - ``"cryptominisat"`` -- note that the cryptominisat package must be
+        - ``"cryptominisat"`` -- note that the pycryptosat package must be
```
Comment (mkoeppe) [inline src/sage/sat/solvers/satsolver.pyx:325]:
```suggestion
        - ``"cryptominisat"`` -- note that the :ref:`pycryptosat <spkg_pycryptosat>` package must be
```

File: build/pkgs/pycryptosat/SPKG.rst [hunk 2]
```diff
@@ -4,8 +4,21 @@ pycryptosat: Python module of cryptominisat
...
+
+   CryptoMiniSat is a SAT solver that aims to become a premiere SAT solver
+   with all the features and speed of successful SAT solvers, such as MiniSat
+   and PrecoSat. The long-term goals of CryptoMiniSat are to be an efficient
+   sequential, parallel and distributed solver. There are solvers that are
+   good at one or the other, e.g. ManySat (parallel) or PSolver (distributed),
+   but we wish to excel at all.
+
+   CryptoMiniSat 2.5 won the SAT Race 2010 among 20 solvers submitted by researchers and industry.
+
```
Comment (mkoeppe) [inline build/pkgs/pycryptosat/SPKG.rst]: Please unindent for proper formatting in the HTML documentation.

Also, the description should just describe the package, not talk about venvs etc.

**Comment by saraedum** [comment]
(dimpase, I took the liberty to fix a typo in the PR summary, hope you don't mind.)

**Comment by dimpase** [comment]
> > that was a fallout of mass file renaming
> 
> No it wasn't. 

I normally don't merge develop, I rebase over it, but this time the rebase was denied by GitHub, for a reason I didn't understand. That was the renamed file, I suppose.

And my foggy post covid brain overlooked it.

File: build/pkgs/pycryptosat/SPKG.rst (hunk 2 above)
Comment (dimpase) [reply build/pkgs/pycryptosat/SPKG.rst]: I haven't added anything about venvs, it was there.

Please edit and add a commit as you see fit.

**Comment by mkoeppe** [comment]
No it wasn't. It was already broken when I reported it in https://github.com/sagemath/sage/pull/37669#issuecomment-2025689778

File: build/pkgs/pycryptosat/SPKG.rst (hunk 2 above)
Comment (dimpase) [reply build/pkgs/pycryptosat/SPKG.rst]: fixed by [6d3d782](https://github.com/sagemath/sage/pull/37669/commits/6d3d782ccf21a07496c35d4b415195fb43d433e2)

**Comment by dimpase** [comment]
ping?

**Comment by mkoeppe** [comment]
OK, let's merge it

**Comment by vbraun** [comment]
merge conflict needs fixing

**Pull request merged**